packages/foundation-models/src/azure-openai/stream/sse-stream.ts

```diff
@@ -0,0 +1,206 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { LineDecoder } from './line-decoder.js';
+import { SSEDecoder } from './sse-decoder.js';
+import type { ServerSentEvent } from './sse-decoder.js';
+import type { HttpResponse } from '@sap-cloud-sdk/http-client';
+
+const logger = createLogger({
+  package: 'foundation-models',
+  messageContext: 'azure-openai-sse-stream'
+});
+
+type Bytes = string | ArrayBuffer | Uint8Array | Buffer | null | undefined;
+
+/**
+ * Stream implemented as an async iterable.
+ * @internal
+ */
+export class SseStream<Item> implements AsyncIterable<Item> {
```

Comment:
General comment, since the complexity of this class is very high, even if multiple functions are only @internal, they should still have proper typedoc.